openapi.yaml

```diff
@@ -0,0 +1,473 @@
+swagger: "2.0"
+info:
+  title: "Reader API"
+  description: "This api service served to end users or third-party apps/services"
+  version: "1.0.0"
+  contact:
+    name: "Gorillab"
+    url: "https://gorillab.co/"
+    email: "hello@gorillab.co"
+host: "localhost:3000"
+basePath: "/api/v1"
+schemes:
+- "http"
+- "https"
+consumes:
+  - "application/json"
+produces:
+  - "application/json"
+tags:
+- name: "auth"
+  description: "Auth module"
+- name: "user"
+  description: "User module"
+- name: "source"
+  description: "Source module"
+- name: "post"
+  description: "Post module"
+paths:
+  /auth/facebook:
+    get:
+      tags:
+      - "auth"
+      summary: "Logs user into the system by facebook"
+      operationId: "loginByFacebook"
+      responses:
+        200:
+          description: "successful operation, send a facebook login form"
```

Comment:
when user clicks on `Login with Facebook` button, it leads to this `url`, and the server redirects to Facebook authentication page. so this should 301 status code.